charts/matrix-stack/templates/matrix-authentication-service/secret_hook.yaml

```
@@ -9,7 +9,7 @@ SPDX-License-Identifier: AGPL-3.0-only
   (or (and $.Values.synapse.checkConfigHook.enabled
         (include "element-io.matrix-authentication-service.synapse-secret-data" (dict "root" $ "context" .))
         )
-      .syn2mas.enabled
+       .syn2mas.enabled (not .syn2mas.dryRun)
```

```suggestion
       (and .syn2mas.enabled (not .syn2mas.dryRun))
```

I think as we're inside the `or` only at this point